src/components/PostsList/PostsList.jsx

```diff
@@ -1,37 +1,60 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { getUserPosts } from '../../api/posts';
+import { Loader } from '../Loader';
+
 import './PostsList.scss';
 
-export const PostsList = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
-
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
-
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
-
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Open
-        </button>
-      </li>
-    </ul>
-  </div>
-);
+export const PostsList = ({ selectedUserId, selectedPostId, selectPost }) => {
+  const [posts, setPosts] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    loadPosts();
+  }, [selectedUserId]);
+
+  const loadPosts = async() => {
+    const loadedPosts = await getUserPosts(selectedUserId);
+
+    setPosts(loadedPosts);
+    setIsLoading(false);
+  };
```

Comment:
let's move the declaration of this func inside useEffect, so it won't be created on each render